json-smart/src/main/java/net/minidev/json/parser/ParserOptions.java

```diff
@@ -0,0 +1,32 @@
+package net.minidev.json.parser;
+
+/*
+ *    Copyright 2011-2023 JSON-SMART authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+public class ParserOptions {
```

Comment:
adding a flag to remove the limit can be enough?

adding and extra ParserOptions only for that, is... overkill ?

and this PR looks not to use maxDepth.

Reply:
@UrielCh Yes, as I said before, it's only a draft about external API changes.